File: code/250/solution.cpp
#include "../template/bigint.cpp"
#include "../template/china_reminder.cpp"
#include "../template/euler.cpp"
#include "../template/pell.cpp"
#include "../template/prime.cpp"
#include "../template/prime_cnt.cpp"
#include <iostream>/*{{{*/
#include <map>
#include <math.h>
#include <set>
#include <vector>
const long long e2 = 100;
const long long e3 = e2 * 10;
const long long e4 = e3 * 10;
const long long e5 = e4 * 10;
const long long e6 = e4 * e2;
const long long e7 = e6 * 10;
const long long e8 = e7 * 10;
const long long e9 = e8 * 10;
long long _mod     = e9 + 7;
using namespace std; /*}}}*/
/*add{{{*/
long long base_add(long long a, long long b, long long mod = _mod) {
    a %= mod;
    b %= mod;
    return (a + b) % mod;
}
long long add(long long a, long long b, long long mod = _mod) { return base_add(a, b, mod); }
long long add(initializer_list<long long> args, long long mod = _mod) {
    long long ret = 0;
    for (auto i : args)
        ret = base_add(ret, i, mod);
    return ret;
}
/*}}}*/
/*{{{mul*/
long long base_mul(long long a, long long b, long long mod = _mod) {
    a %= mod;
    b %= mod;
    long long ret = 0;
    while (b) {
        if (b & 1)
            ret = (ret + a) % mod;
        b >>= 1;
        a = (a + a) % mod;
    }
    return ret;
}
long long mul(long long a, long long b, long long mod = _mod) { return base_mul(a, b, mod); }
long long mul(initializer_list<long long> args, long long mod = _mod) {
    long long ret = 1;
    for (auto i : args)
        ret = base_mul(ret, i, mod);
    return ret;
} /*}}}*/
/*{{{qpow*/
long long qpow(long long a, long long b, long long mod = _mod) {
    long long ret = 1;
    while (b) {
        if (b & 1)
            ret = mul(ret, a, mod);
        a = mul(a, a, mod);
        b >>= 1;
    }
    return ret;
} /*}}}*/

long long num[300];
long long cc[300];
long long mod = e8 * e8;
int main() {
    _mod   = 250;
    num[0] = 1;
    for (int i = 1; i <= 250250; i++) {
        int now = qpow(i, i);
        for (int i = 0; i < 250; i++)
            cc[i] = 0;
        for (int i = 0; i < 250; i++) {
            auto &x = cc[add(i, now)];
            x       = (x + num[i]) % mod;
        }
        for (int i = 0; i < 250; i++) {
            num[i] = (num[i] + cc[i]) % mod;
        }
    }
    cout << num[0] - 1 << endl;
    return 0;
}

File: code/250/solution.go
package main /*{{{*/

import (
	"fmt"
	"math"
	"reflect"
)

const (
	e4  = int64(10000)
	e5  = int64(100000)
	e6  = int64(1000000)
	e7  = int64(10000000)
	e8  = e4 * e4
	mod = int64(250)
)

func sqrt(a int64) int64 {
	x := math.Sqrt(float64(a))
	return int64(x)
}

func add(list ...interface{}) int64 { /*{{{*/
	if len(list) == 0 {
		return 0
	}
	ret := int64(0)
	for _, i := range list {
		buf := int64(0)
		switch reflect.TypeOf(i).String() {
		case "int":
			buf = int64(i.(int))
		case "int64":
			buf = i.(int64)
		}
		ret = (ret + buf) % mod
	}
	return ret
} /*}}}*/
func mul(list ...interface{}) int64 { /*{{{*/
	if len(list) == 0 {
		return 0
	}
	ret := int64(1)
	for _, i := range list {
		buf := int64(0)
		switch reflect.TypeOf(i).String() {
		case "int":
			buf = int64(i.(int))
		case "int64":
			buf = i.(int64)
		}
		buf %= mod
		base := ret
		ret = 0
		for buf > 0 {
			if (buf & 1) == 1 {
				ret = add(ret, base)
			}
			base = add(base, base)
			buf >>= 1
		}
	}
	return ret
} /*}}}*/
func qpow(a, b int64) int64 { /*{{{*/
	ret := int64(1)
	for b > 0 {
		if (b & 1) == 1 {
			ret = mul(ret, a)
		}
		a = mul(a, a)
		b >>= 1
	}
	return ret
} /*}}}*/
/*}}}*/

var num = [300]int64{}
var cc = [300]int64{}
var mod2 = e8 * e8

func main() {
	num[0] = 1
	for i := int64(1); i <= 250250; i++ {
		now := qpow(i, i)
		cc = [300]int64{}
		for i := 0; i < 250; i++ {
			cc[add(i, now)] = (cc[add(i, now)] + num[i]) % mod2
		}
		for i := 0; i < 250; i++ {
			num[i] = (num[i] + cc[i]) % mod2
		}
	}
	fmt.Println(num[0] - 1)
}
